src/java/org/apache/cassandra/repair/autorepair/AutoRepair.java

```diff
@@ -566,4 +569,35 @@ public void progress(String tag, ProgressEvent event)
             }
         }
     }
+
+    public synchronized void shutdownBlocking() throws ExecutionException, InterruptedException
+    {
+        if (!isSetupDone)
+        {
+            // By default, executors within AutoRepair are not initialized as the feature is opt-in.
+            // If the AutoRepair has not been set up, then there is no need to worry about shutting it down
+            return;
+        }
+        if (isShutDown)
+        {
+            throw new IllegalStateException("AutoRepair has already been shut down");
+        }
```

Comment:
This isn't super important and is more of a personal preference, but could we make a subsequent shutdown a no op instead like this:

```suggestion
        if (!isSetupDone || isShutdown)
        {
            // By default, executors within AutoRepair are not initialized as the feature is opt-in.
            // If the AutoRepair has not been set up, then there is no need to worry about shutting it down
            return;
        }
```

I see that some things in the shutdown path `HintsService.shutdownBlocking` would throw on subsequent calls, while others don't (e.g. `CommitLog.instance.shutdownBlocking`).   I like the behavior of not throwing an exception, which is similar to how java's `Executor.shutdown()` behaves.

Reply:
actually, given `HintsService.shutdownBlocking` behaves this way, I see no harm in leaving it this way.  Given the behavior is inconsistent between different components being shut down, I think there isn't much value in changing the behavior unless we make everything behave the same.  e.g. if `drain` gets called twice, other components would throw an exception before this is even invoked.

Since we already have a relatively clean test run, I'm going to give +1 so we can get this merged right on in.